Emby.Server.Implementations/HttpServer/HttpResultFactory.cs

```diff
@@ -348,13 +348,19 @@ private object ToOptimizedResultInternal<T>(IRequest request, T dto, IDictionary
         private byte[] Compress(byte[] bytes, string compressionType)
         {
             if (string.Equals(compressionType, "br", StringComparison.OrdinalIgnoreCase))
```

Comment:
cannot this be done with a `switch` thing?

Reply:
See Riot ;)

Reply:
Those benchmarks are all in nanoseconds, so performance here is a moot point IMHO.

Compare those two versions **for readability**:

_yours_:
```cs
        private byte[] Compress(byte[] bytes, string compressionType)
        {
            if (string.Equals(compressionType, "br", StringComparison.OrdinalIgnoreCase))
            {
                return CompressBrotli(bytes);
            }

            if (string.Equals(compressionType, "deflate", StringComparison.OrdinalIgnoreCase))
            {
                return Deflate(bytes);
            }

            if (string.Equals(compressionType, "gzip", StringComparison.OrdinalIgnoreCase))
            {
                return GZip(bytes);
            }

            throw new NotSupportedException(compressionType);
        }
```

_mine_:
```cs
        private byte[] Compress(byte[] bytes, string compressionType)
        {
            switch (compressionType.ToLowerInvariant()) {
                case "br": return CompressBrotli(bytes);
                case "deflate": return Deflate(bytes);
                case "gzip": return GZip(bytes);
            }

            throw new NotSupportedException(compressionType);
        }
```

Reply:
Even we chose to (I'm not convinced) this is out of the scope of this PR. I didn't even touch the logic.

Reply:
Well, eventually all this should be replaced by Kestrel or something similar, so yeah, out of scope of this PR.